src/modules/job-ingest/job-ingest.c

```diff
@@ -117,6 +118,14 @@ struct batch {
     json_t *joblist;
 };
 
+struct batch_response {
+    flux_future_t *f;
+    bool complete_failure;
```

Comment:
nit: `job-info` uses the term `completed` for jobs that have finished running.  perhaps use a different flag name just to avoid potential confusion? `all_batch_failed` or similar?